modules/images/image-loading-optimization/storage/rest-api.php

```diff
@@ -171,16 +172,28 @@ function ilo_handle_rest_request( WP_REST_Request $request ) {
 		ilo_get_url_metrics_breakpoint_sample_size(),
 		ilo_get_url_metric_freshness_ttl()
 	);
-	if ( ! ilo_needs_url_metric_for_breakpoint( $needed_minimum_viewport_widths ) ) {
+
+	// Block the request if URL metrics aren't needed for the provided viewport width.
+	// This logic is the same as the isViewportNeeded() function in detect.js.
+	$viewport_width  = $request->get_param( 'viewport' )['width'];
+	$last_was_needed = false;
+	foreach ( $needed_minimum_viewport_widths as list( $minimum_viewport_width, $is_needed ) ) {
+		if ( $viewport_width >= $minimum_viewport_width ) {
+			$last_was_needed = $is_needed;
+		} else {
+			break;
+		}
+	}
+	if ( ! $last_was_needed ) {
```

Comment:
Is the `ilo_needs_url_metric_for_breakpoint()` still valuable given it's no longer used here? Would it be possible to find a way to enhance the function in a way that it can be used here, rather than more inline logic in this function?

Reply:
Good point! Yeah, it's not valuable. I've removed it in 3da1363c. I did try to consolidate this logic into that existing function, but the two codepaths didn't mesh well.